src/ClientData/include/ClientData/TimerData.h

```diff
@@ -23,11 +23,27 @@ class TimerData final : public TimerDataInterface {
   // Timers queries
   [[nodiscard]] std::vector<const TimerChain*> GetChains() const override;
   [[nodiscard]] const TimerChain* GetChain(uint64_t depth) const;
+
+  // The method is not optimized. The complexity is linear in the total number of timer_infos,
+  // sortedness is not made use of.
   [[nodiscard]] virtual std::vector<const orbit_client_protos::TimerInfo*> GetTimers(
-      uint64_t /*min_tick*/ = std::numeric_limits<uint64_t>::min(),
-      uint64_t /*max_tick*/ = std::numeric_limits<uint64_t>::max()) const override {
-    // TODO(b/204173236): Implement GetTimers and use it in TimerTracks.
-    return {};
+      uint64_t min_tick = std::numeric_limits<uint64_t>::min(),
+      uint64_t max_tick = std::numeric_limits<uint64_t>::max()) const override {
+    // TODO(b/204173236): use it in TimerTracks.
+    std::vector<const orbit_client_protos::TimerInfo*> timers;
+    absl::MutexLock lock(&mutex_);
```

Comment:
As much as the declaration of `timers` doesn't need to be under the mutex, I think it's good practice to lock the mutex at the very beginning of the function when it's supposed to conceptually cover the whole function, so that it's immediately visible.